include/clad/Differentiator/BuiltinDerivatives.h

```
@@ -72,6 +72,16 @@ CUDA_HOST_DEVICE ValueAndPushforward<T, T> sqrt_pushforward(T x, T d_x) {
   return {::std::sqrt(x), (((T)1) / (((T)2) * ::std::sqrt(x))) * d_x};
 }
 
+template <typename T>
+CUDA_HOST_DEVICE ValueAndPushforward<T, T> floor_pushforward(T x, T /*d_x*/) {
+  return {::std::floor(x), (T)0};
+}
+
+template <typename T>
+CUDA_HOST_DEVICE ValueAndPushforward<T, T> ceil_pushforward(T x, T /*d_x*/) {
```

Same thing for ceil function. Derivative should be undefined for integers. Also, the derivative function seems entirely wrong when `T` is an integral type.

---

@PetroZarytskyi ping.

---

Basically, when I proposed adding derivatives for ceil and floor I thought we assumed the function is differentiable (i.e. we didn't care about discontinuity conditions and loops may introduce). However, the fact that the derivative of ceil/floor is undefined doesn't mean that the derivative of the outer function is undefined so I don't think we should throw an error. A warning would make sense for sure.

---

Also, in this case, we should take into account that the derivative of abs is undefined at 0.